neutone_sdk/audio.py

```diff
@@ -127,14 +132,47 @@ def render_audio_sample(
     audio_len = audio.size(1)
     padding_amount = math.ceil(audio_len / buffer_size) * buffer_size - audio_len
     padded_audio = nn.functional.pad(audio, [0, padding_amount])
-    chunks = padded_audio.split(buffer_size, dim=1)
+    audio_chunks = padded_audio.split(buffer_size, dim=1)
 
     model.set_daw_sample_rate_and_buffer_size(
         preferred_sr, buffer_size, preferred_sr, buffer_size
     )
-    audio_out = tr.hstack(
-        [model.forward(chunk, params).clone() for chunk in tqdm(chunks)]
-    )[:, :audio_len]
+
+    # make sure the shape of params is compatible with the model calls.
+    if params is not None:
+        assert params.shape[0] == model.MAX_N_PARAMS
+
+        # if constant values, copy across audio dimension
+        if params.dim() == 1:
+            params = params.repeat([audio_len, 1]).T
+
+        # otherwise resample to match audio
+        else:
+            assert params.shape == (model.MAX_N_PARAMS, input_sample.audio.size(1))
+            params = torchaudio.transforms.Resample(input_sample.sr, preferred_sr)(
+                params
+            )
+            params = torch.clamp(params, 0, 1)
+
```

Comment:
Do you mind running the black linter (https://black.readthedocs.io/en/stable/getting_started.html#basic-usage) on this file real quick just to keep the formatting consistent?